[daikin] NPE on startup

On openhab 3.2.0 startup, I saw the following error:

```
19:47:36.475 [ERROR] [rnal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.handleCommand()' on 'org.openhab.binding.daikin.internal.handler.DaikinAirbaseUnitHandler@11d10ec7': null
java.lang.NullPointerException: null
	at org.openhab.binding.daikin.internal.handler.DaikinAirbaseUnitHandler.changeZone(DaikinAirbaseUnitHandler.java:181) ~[?:?]
	at org.openhab.binding.daikin.internal.handler.DaikinAirbaseUnitHandler.handleCommandInternal(DaikinAirbaseUnitHandler.java:70) ~[?:?]
	at org.openhab.binding.daikin.internal.handler.DaikinBaseHandler.handleCommand(DaikinBaseHandler.java:97) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:154) [bundleFile:?]
	at org.openhab.core.internal.common.InvocationHandlerSync.invoke(InvocationHandlerSync.java:59) [bundleFile:?]
	at com.sun.proxy.$Proxy237.handleCommand(Unknown Source) [?:?]
	at org.openhab.core.thing.internal.profiles.ProfileCallbackImpl.handleCommand(ProfileCallbackImpl.java:80) [bundleFile:?]
	at org.openhab.core.thing.internal.profiles.SystemDefaultProfile.onCommandFromItem(SystemDefaultProfile.java:48) [bundleFile:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:154) [bundleFile:?]
	at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
```

I believe this is because `airbaseModelInfo` isn't yet populated by the `pollStatus()` and a rule on startup requests a change of zone.

--- a/bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java
+++ b/bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java
@@ -72,7 +72,6 @@ public class DaikinAcUnitHandler extends DaikinBaseHandler {
             return;
         }
         ControlInfo controlInfo = webTargets.getControlInfo();
-        updateStatus(ThingStatus.ONLINE);
         updateState(DaikinBindingConstants.CHANNEL_AC_POWER, controlInfo.power ? OnOffType.ON : OnOffType.OFF);
         updateTemperatureChannel(DaikinBindingConstants.CHANNEL_AC_TEMP, controlInfo.temp);
 
@@ -153,6 +152,7 @@ public class DaikinAcUnitHandler extends DaikinBaseHandler {
             // Suppress any error if energy info is not supported.
             logger.debug("getEnergyInfoDayAndWeek() error: {}", e.getMessage());
         }
+        updateStatus(ThingStatus.ONLINE);
     }
 
     @Override

--- a/bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAirbaseUnitHandler.java
+++ b/bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAirbaseUnitHandler.java
@@ -77,7 +77,6 @@ public class DaikinAirbaseUnitHandler extends DaikinBaseHandler {
     @Override
     protected void pollStatus() throws IOException {
         AirbaseControlInfo controlInfo = webTargets.getAirbaseControlInfo();
-        updateStatus(ThingStatus.ONLINE);
 
         if (airbaseModelInfo == null || !"OK".equals(airbaseModelInfo.ret)) {
             airbaseModelInfo = webTargets.getAirbaseModelInfo();
@@ -115,6 +114,7 @@ public class DaikinAirbaseUnitHandler extends DaikinBaseHandler {
                     .forEach(idx -> updateState(DaikinBindingConstants.CHANNEL_AIRBASE_AC_ZONE + idx,
                             OnOffType.from(zoneInfo.zone[idx])));
         }
+        updateStatus(ThingStatus.ONLINE);
     }
 
     @Override
@@ -178,18 +178,25 @@ public class DaikinAirbaseUnitHandler extends DaikinBaseHandler {
     }
 
     protected void changeZone(int zone, boolean command) throws DaikinCommunicationException {
-        if (zone <= 0 || zone > airbaseModelInfo.zonespresent) {
+        AirbaseZoneInfo zoneInfo = webTargets.getAirbaseZoneInfo();
+        long commonZones = 0;
+        long maxZones = zoneInfo.zone.length;
+
+        if (airbaseModelInfo != null) {
+            maxZones = Math.min(maxZones - 1, airbaseModelInfo.zonespresent);
+            commonZones = airbaseModelInfo.commonzone;
+        }
+        if (zone <= 0 || zone > maxZones) {
             logger.warn("The given zone number ({}) is outside the number of zones supported by the controller ({})",
-                    zone, airbaseModelInfo.zonespresent);
+                    zone, maxZones);
             return;
         }
 
-        AirbaseZoneInfo zoneInfo = webTargets.getAirbaseZoneInfo();
-        long count = IntStream.range(0, zoneInfo.zone.length).filter(idx -> zoneInfo.zone[idx]).count()
-                + airbaseModelInfo.commonzone;
-        logger.debug("Number of open zones: \"{}\"", count);
+        long openZones = IntStream.range(0, zoneInfo.zone.length).filter(idx -> zoneInfo.zone[idx]).count()
+                + commonZones;
+        logger.debug("Number of open zones: \"{}\"", openZones);
 
-        if (count >= 1) {
+        if (openZones >= 1) {
             zoneInfo.zone[zone] = command;
             webTargets.setAirbaseZoneInfo(zoneInfo);
         }

